### PR TITLE
Add method for extracting walls from Revit models

### PR DESCRIPTION
**Pull request by wynged**
BACKGROUND:
We want to export walls from Revit as well as floors.

DESCRIPTION:
This PR adds nodes to extract walls, there is a new UserElement class created to make the wall from a face extracted from the Revit wall.

TESTING:
run `dotnet build` and checkout the new nodes.
  
FUTURE WORK:
This only handles Planar walls with a line centerline.

COMMENTS:
There are some geometry issues around the intersections of walls, the joining behavior is not kept with this method of gathering the wall geometry.

---
This change is [Reviewable](https://reviewable.io/reviews/hypar-io/elements/246)

### CODE REVIEW

File: src/Revit/RevitHyparTools/Create.Wall.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using Autodesk.Revit.DB;
+using Elements.Geometry;
+using GeometryEx;
+using ElemGeom = Elements.Geometry;
+
+using Revit = Autodesk.Revit.DB;
+
+namespace RevitHyparTools
```
Comment (ikeough) [inline src/Revit/RevitHyparTools/Create.Wall.cs]: `namespace Hypar.Revit`

File: src/Dynamo/src/HyparDynamo/HyparDynamo.csproj [hunk 2]
```diff
@@ -31,14 +30,15 @@
 
   <ItemGroup>
     <ProjectReference Include="..\..\..\Revit\RevitHyparTools\RevitHyparTools.csproj" />
+    <ProjectReference Include="..\..\..\Elements\Elements.csproj" />
```
Comment (ikeough) [inline src/Dynamo/src/HyparDynamo/HyparDynamo.csproj]: This should not be a local reference right? This should point to `Elements`?

File: src/Revit/RevitHyparTools/Create.Wall.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+{
+    public static partial class Create
```
Comment (ikeough) [inline src/Revit/RevitHyparTools/Create.Wall.cs]: Remind me why this is `partial`.

File: src/Revit/RevitHyparTools/RevitHyparTools.csproj [hunk 4]
```diff
@@ -16,4 +15,7 @@
     </Reference>
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="..\..\Elements\Elements.csproj" />
```
Comment (ikeough) [inline src/Revit/RevitHyparTools/RevitHyparTools.csproj]: No local reference as per comment above.

File: src/Revit/RevitHyparTools/WallByProfile.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Elements;
+using Elements.Geometry;
+using Elements.Geometry.Solids;
+
+namespace RevitHyparTools {
+    [UserElement]
+    public class WallByProfile : Wall
```
Comment (ikeough) [inline src/Revit/RevitHyparTools/WallByProfile.cs]: Always add `///` comments to the class and to its methods and fields.

File: src/Revit/RevitHyparTools/Create.Wall.cs (hunk 1 above)
Comment (wynged) [reply src/Revit/RevitHyparTools/Create.Wall.cs]: ok, i'll do this here and other places

File: src/Dynamo/src/HyparDynamo/HyparDynamo.csproj (hunk 2 above)
Comment (wynged) [reply src/Dynamo/src/HyparDynamo/HyparDynamo.csproj]: all this work is inside the Elements repo, and this reference points to the Elements.csproj inside the Elements repo that this project is in.... 
a bit confusing but basically I'm point to an "adjacent" elements.csproj in the same repo, which I think is right, this should always be using the latest elements no?

File: src/Revit/RevitHyparTools/Create.Wall.cs (hunk 3 above)
Comment (wynged) [reply src/Revit/RevitHyparTools/Create.Wall.cs]: the pattern I have is that there is one Create class spread across multiple files depending on what is being created.  It is more obvious in subsequent work with Floor and columns following the same pattern

File: src/Revit/RevitHyparTools/RevitHyparTools.csproj (hunk 4 above)
Comment (wynged) [reply src/Revit/RevitHyparTools/RevitHyparTools.csproj]: same thought,  I feel like we should reference the version of Elements that is in this repo, not the nuget version.

File: src/Dynamo/src/HyparDynamo/HyparDynamo.csproj (hunk 2 above)
Comment (ikeough) [reply src/Dynamo/src/HyparDynamo/HyparDynamo.csproj]: I see now. The way you’ve done it is correct.

File: src/Revit/RevitHyparTools/WallByProfile.cs (hunk 5 above)
Comment (wynged) [reply src/Revit/RevitHyparTools/WallByProfile.cs]: this is moving to be a generated type, so I'm not going to do this.

**Pull request merged**